src/mongocrypt.h

```diff
@@ -1188,6 +1188,23 @@ bool mongocrypt_kms_ctx_feed(mongocrypt_kms_ctx_t *kms, mongocrypt_binary_t *byt
 MONGOCRYPT_EXPORT
 bool mongocrypt_kms_ctx_fail(mongocrypt_kms_ctx_t *kms);
 
+/**
+ * Reset a KMS context allowing it to be retried.
```

Comment:
I think this comment could be expanded a bit regarding the specific situations in which this method should be called and what exactly "reset" means w.r.t. the state of the `kms` handle?